src/psyclone/psyir/nodes/intrinsic_call.py

```diff
@@ -125,7 +125,12 @@ def __init__(self, routine, **kwargs):
                 f"'{type(routine).__name__}'.")
 
         # A Call expects a symbol, so give it an intrinsic symbol.
-        super().__init__(IntrinsicSymbol(routine.name), **kwargs)
+        super().__init__(
+            IntrinsicSymbol(
+                routine.name,
+                is_elemental=(routine in ELEMENTAL_INTRINSICS),
+                is_pure=True),
```

Comment:
Are all intrinsic pure?
In fact what the ones that we have: allocate, deallocate, random_number are not.
Maybe it should be 'None' for now?

Reply:
I changed my mind and now set it correctly (I hope).